### ISSUE TITLE
[DR] Enable cuDSS-based linear solvers (on GPU)

### ISSUE THREAD

**Issue by manauref** [ammarhakim/gkylzero#385]
Presently we can't compile with cuda >=12.2 without getting lots of warnings about some cuSparse features being deprecated, and when you try to run our cuSolver-based linear solvers unit test, or GK reg tests, they seg fault (see issue #234 ).

On solution proposed here is to solve linear problems
A_i x_i = b_i where i=1,2,...,Nprob
or
A x_j = b_j where j=1,2,...,Nrhs
using [cuDSS](https://developer.nvidia.com/cudss) instead.

The idea is to:
1. Allow users to specify via machine-file flags (e.g. --use-cudss=yes) whether to use cuDSS. When true, gkylzero will download cuDSS into gkylsoft and link to it. This also sets GKYL_HAVE_CUDSS=1
2. Create a zero/cudss_ops.c module that has the same functionality (and hopefully more) functionality that's in cusolver_ops.c
3. Then updaters, like fem_poisson, can choose do perform certain steps with cuSolver or cuDSS based on `GKYL_HAVE_CUDSS=1` like (see around L197 of zero/fem_poisson.c)
```
  if (up->use_gpu) {
#ifdef GKYL_HAVE_CUDSS
    gkyl_cudss_amat_from_triples(up->prob_cu, tri);
#else
    gkyl_cusolver_amat_from_triples(up->prob_cu, tri);
#endif
  }
```
Eventually we may deprecate one of these two solvers, depending on NVIDIA's recommendations and maintenance schedules, and our users experience.

For now we focus on problems where A is time independent. I believe cuDSS will allow us to do the time dependent problem A=A(t) as well (e.g. for time dependent polarization density weights), which wasn't possible with cuSolver. Other benefits of cuDSS include improved performance, more operations happening on the GPU (cuSolver did some things on the host), and potentially better support within NVIDIA.

This is currently being prototyped in the `gk-g0-app_cudss-prototyping` branch, where an initial, simple version of zero/cudss_ops.c has been added, along with a unit test.

**Comment by ammarhakim** [comment]
Comments:

- Move the ifdef preprocessor to the method themselves and not in each usage. So a common method called "cuda_nmat" and not "cusolver_nmat". For example.
- Replace printf to fprintf and use stderror file pointer.

**Comment by manauref** [comment]
Addressed in PR #396
